drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/scan_card_loader.rb

```diff
@@ -0,0 +1,78 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisExternalApis::TcHmis::Importers::Loaders
+  class ScanCardLoader < BaseLoader
+    def initialize(...)
+      super
+
+      @client_lookup = Hmis::Hud::Client.
+        where(data_source_id: data_source.id).
+        pluck(:personal_id, :id).
+        to_h
+
+      @now = Time.current
+    end
+
+    def filename
+      'ScanCodes.xlsx'
+    end
+
+    # why not define this in BaseLoader?
+    def runnable?
+      super && reader.file_present?(filename)
+    end
+
+    def perform
+      Hmis::ScanCardCode.with_deleted.each(&:really_destroy!) if clobber
+
+      actual = 0
+      expected = 0
+      scan_cards = []
+      unique_codes = Set.new
+
+      rows = reader.rows(filename: filename)
+      rows.each do |row|
+        case_number = row.field_value_by_id('Case Number')
+        next unless case_number && case_number.length > 1
+        next unless case_number.starts_with?('P') # not expected, file should already filter them out
+
+        expected += 1
```

Comment:
maybe this should go before line 39 so we know how many blanks or invalid rows we are skipping